src/ode_solver/low_storage_runge_kutta_ode_solver.cpp

```diff
@@ -0,0 +1,258 @@
+                this->dg->fe_collection,
+                this->dg->volume_quadrature_collection,
+                this->dg->high_order_grid->fe_system.tensor_degree(),
+                this->dg->max_degree,
+                this->dg->oneD_fe_collection_1state,
+                this->dg->oneD_quadrature_collection);
+        }
+        this->dg->assemble_residual();
+        this->dg->apply_inverse_global_mass_matrix(this->dg->right_hand_side, rhs);
+        rhs *= dt;
+        storage_register_4.add(this->butcher_tableau->get_b_hat(n_rk_stages), rhs);
+
+    }
+    this->dg->solution = storage_register_1;
+
+    // Apply limiter at every RK stage
+    if (this->limiter) {
+        this->limiter->limit(this->dg->solution,
+            this->dg->dof_handler,
+            this->dg->fe_collection,
+            this->dg->volume_quadrature_collection,
+            this->dg->high_order_grid->fe_system.tensor_degree(),
+            this->dg->max_degree,
+            this->dg->oneD_fe_collection_1state,
+            this->dg->oneD_quadrature_collection);
+        }
+ 
+    this->pcout << std::endl;
+    ++(this->current_iteration);
+    this->current_time += dt;
+    this->pcout << " Time is: " << this->current_time <<std::endl;
+    this->pcout << std::endl;
+}
+
+template <int dim, typename real, int n_rk_stages, typename MeshType> 
+double LowStorageRungeKuttaODESolver<dim,real,n_rk_stages, MeshType>::get_automatic_error_adaptive_step_size (real dt, const bool /*pseudotime*/)
+{
+    double error = 0.0;
+    w = 0.0;
+
+    // error based step size 
+    if (is_3Sstarplus == false){
+        // loop sums elements at each mpi processor
+        for (dealii::LinearAlgebra::distributed::Vector<double>::size_type i = 0; i < storage_register_1.local_size(); ++i) {
+            error = storage_register_1.local_element(i) - storage_register_2.local_element(i);
+            w = w + pow(error / (atol + rtol * std::max(std::abs(storage_register_1.local_element(i)), std::abs(storage_register_2.local_element(i)))), 2);
+        }
+    }
+    else if (is_3Sstarplus == true){
+        // loop sums elements at each mpi processor
+        for (dealii::LinearAlgebra::distributed::Vector<double>::size_type i = 0; i < storage_register_1.local_size(); ++i) {
+            error = storage_register_1.local_element(i) - storage_register_4.local_element(i);
+            w = w + pow(error / (atol + rtol * std::max(std::abs(storage_register_1.local_element(i)), std::abs(storage_register_4.local_element(i)))), 2);
+        }
+    }
+    this->pcout << std::endl;
+
+    // sum over all elements
+    w = dealii::Utilities::MPI::sum(w, this->mpi_communicator);
+    w = pow(w / global_size, 0.5);
+    epsilon[2] = epsilon[1];
+    epsilon[1] = epsilon[0];
+    epsilon[0] = 1.0 / w;
+    dt = pow(epsilon[0], 1.0 * beta1/rk_order) * pow(epsilon[1], 1.0 * beta2/rk_order) * pow(epsilon[2], 1.0 * beta3/rk_order) * dt;
+    this->pcout << " dt: " << dt;
```

Comment:
There is a parameter `ode_param.ode_output` which could be used to control whether debugging information is printed. See this [line](https://github.com/dougshidong/PHiLiP/blob/5ec59ea934e6577aa7c71ef64f64602ee8e29ab8/src/ode_solver/implicit_ode_solver.cpp#L31) for example. Could you add a similar structure to control debug printing?

It's completely fine to just use `this->pcout` without checking the output mode if you're still doing personal testing/debugging, but using the `ode_output` param will be helpful when this is in master!